### PR TITLE
Update go bindings version to the rarimo/evm-bridge-contracts@bindings, refactor rpc query usage to one core client, resolve some todos

### PR DESCRIPTION
**Pull request by napalmpapalam**
### Changed
- `Transfer[token]` resource property renamed to the `Transfer[item]`
- `/tokens` endpoint path renamed to the `/items`
- `token_index` path parameter for the Balance and NFTMetadata endpoints renamed to `index`
- `token_index` query parameter for the Transfers endpoints renamed to `item_index`
- Migrated to the `rarimo/evm-bridge-contracts@bindinds` autogenerated bindings
- `token_index` column in the `transfers` table renamed to `item_index`
- `Core` interface extended with the new methods and refactored all occurrences where it wasn't used

### Removed
- ChainGateway interface
- NFTs endpoint
- Unused dependencies from the ApprovalIndexer and RejectionIndexer
- NFTS and TOKENS resources

### Fixed
- Using cached storage for the `TransferByID` and `Transfers` endpoints

### CODE REVIEW

**Comment by ihordiachenko** [comment]
@napalmpapalam reflect the changes in the changelog

File: docs/spec/components/schemas/Transfer.yaml [hunk 1]
```diff
@@ -57,12 +57,12 @@ allOf:
         type: object
         required: [creator]
         properties:
-          token:
+          item:
```
Comment (ihordiachenko) [inline docs/spec/components/schemas/Transfer.yaml:60]: This is a breaking change. Are there any clients depending on the old version?

File: internal/services/api/handlers/balance.go [hunk 2]
```diff
@@ -109,13 +109,20 @@ func Balance(w http.ResponseWriter, r *http.Request) {
...
 
-	ape.Render(w, newBalanceResponse(balance))
+	ape.Render(w, newBalanceResponse(req, balance))
 }
 
-func newBalanceResponse(amount *amount.Amount) resources.BalanceResponse {
+func newBalanceResponse(request *balanceRequest, amount *amount.Amount) resources.BalanceResponse {
+	tokenID := request.TokenID
+	if tokenID == "" {
+		tokenID = "0"
```
Comment (ihordiachenko) [inline internal/services/api/handlers/balance.go:118]: Shouldn't it be hexdecimal?

File: docs/spec/components/schemas/Transfer.yaml (hunk 1 above)
Comment (napalmpapalam) [reply docs/spec/components/schemas/Transfer.yaml:60]: only horizon https://github.com/rarimo/relayer-svc/pull/2

**Review by ihordiachenko — APPROVED** [review]

**Pull request merged**